ankimorphs/text_highlighting.py

```diff
@@ -1,238 +1,377 @@
+
+                ruby = None
+                stat = None
+                continue
+
+            # If the ruby starts then status starts, ruby ends, status ends
+            #
+            if ruby.start < stat.start and ruby.end < stat.end:
+                # print("The ruby starts then status starts, ruby ends, status ends.")
+                self._highlighted = (
+                    self._highlighted[: ruby.start]
+                    + ruby.open()
+                    + self._highlighted[ruby.start : stat.start]
+                    + stat.open()
+                    + self._highlighted[
+                        stat.start : stat.start
+                        + (stat.end - stat.start)
+                        - (stat.end - ruby.end)
+                    ]
+                    + stat.close()
+                    + ruby.rt()
+                    + ruby.close()
+                    + stat.open()
+                    + self._highlighted[ruby.end : stat.end]
+                    + stat.close()
+                    + self._highlighted[stat.end :]
+                )
+
+                stat = None
+
+                # Pull and process statuses until the next status is outside of this ruby.
+                #
+                while self.statuses:
+                    if self.statuses[-1].end <= ruby.start:
+                        break
+
+                    stat = self.statuses.pop()
+                    stat.start += ruby.prefix_len()
+                    stat.end += ruby.prefix_len()
+                    self._highlighted = stat.inject(self._highlighted)
+
+                ruby = None
+                stat = None
+                continue
+
+            # If the status starts then ruby starts, status ends, ruby ends
+            #
+            if ruby.start > stat.start and ruby.end > stat.end:
+                # print("The status starts then ruby starts, status ends, ruby ends.")
+                self._highlighted = (
+                    self._highlighted[: stat.start]
+                    + stat.open()
+                    + self._highlighted[stat.start : ruby.start]
+                    + stat.close()
+                    + ruby.open()
+                    + stat.open()
+                    + self._highlighted[ruby.start : ruby.start + ruby.rt_offset()]
+                    + stat.close()
+                    + self._highlighted[stat.end : ruby.end]
+                    + ruby.rt()
+                    + ruby.close()
+                    + self._highlighted[ruby.end :]
+                )
+                ruby = None
+                stat = None
+                continue
+
+            # print("errrr what the what?")
+
+
+class MorphemeMeta:
+    """A class to track morpheme data relevant to highlighting."""
+
+    def __init__(self, morpheme: Morpheme, am_config: AnkiMorphsConfig):
+        self.text = (
+            morpheme.inflection
+            if am_config.evaluate_morph_inflection
+            else morpheme.lemma
+        ).lower()
```

Comment:
@mortii This is were I determine wether to use inflection or lemma for highlighting. However, based on your comment I assume this is not correct. I'll look into `am_config.evaluate_morph_inflection` further to better understand it.

Reply:
I think it would work if you just set the text to always be the inflection. I think only wrapping the could be a little confusing.